arch-design/api/UnifiedAPI.md

```diff
@@ -0,0 +1,208 @@
+# Unified API for SODA Projects
+
+Author(s): [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Motivation and background
+
+SODA is a collection of multiple independent projects. Each project has their own API server and exposes different endpoints. A single project may be able to handle a specific use case but to handle a large set of use-cases, few projects need to be plugged together. And there should be a single endpoint for all projects, so that clients need not to configure multiple endpoints for SODA projects.
+
+### Goals
+
+The main goal is to provide a unified API and single endpoint access for all projects of SODA. The current scope is limited to unification of hotpot and multi-cloud projects API.
+
+### Non-Goals
+
+The current proposal is not focusing on other projects of SODA, Ex: delfin etc.
+
+### Requirement Analysis
+
+##### Functional Requirements
+
+1. A single entity should contain all the SODA projects API. In other words, Unification of API of all below projects.
+
+  [Hotpot API](https://github.com/sodafoundation/api/blob/master/openapi-spec/swagger.yaml)
+
+  [Multi-cloud API](https://github.com/sodafoundation/multi-cloud/blob/master/openapi-spec/swagger.yaml)
+
+2. All the SODA projects should be access from single endpoint
+
+3. All API should return return-code or/and response messages.
+
+4. All the existing features of both projects should be working from Unified API-Server
+
+5. A single swagger should there for Unified APIs
+
+
+##### Non Functional Requirements
+
+1. The system should be highly available. Multiple users can login to the system and provision resources like, create /volume/fileshare/buckets etc.
+
+2. The system should work with minimum latency.
+
+3. The system should be scalable and it should handle the high amount of users' requests.
+
+## Proposal
+
+To support Unified API for SODA projects, all other project's API can be added to one project named API-Server. Currently API-Server has Hotpot(on-prem storage) related APIs, now other projects like multi-cloud API can also be added to API-Server project which inherently will become similar to API gateway for SODA projects. Further all the request calls coming to API-Server can be validated and forwarded to the Controller project.
+
+Controller is a project, where all the intelligence logics related to selection of on-prem backend storages are there. Basically, this selection logic helps to pick one storage among multiple registered storages based on profile. Later that selected storage can be used for provisioning etc. Now the selection logic can be enhanced to select one storage for provisioning among on-prem and cloud storages. Here selection logic can take input from profile and tiers.
```

Comment:
This seems to be a very specific use case. Suppose there is one API for storage usage data. Now, this data can be consolidated and come from different projects or different platforms like on-prem/cloud/edge. How will this be handled?